src/policy/compiler.rs

```diff
@@ -1397,18 +1412,27 @@ mod tests {
             assert_eq!(small_thresh_ms, small_thresh_ms_expected);
         }
 
-        // Above 20 keys, thresh is always compiled to a ms thresh no matter the value of k
-        // FIXME: k = 1 doesn't work !
+        // Above 20 keys, thresh is compiled to a combination of and()s if it's a N of N,
+        // and to a ms thresh otherwise.
+        // FIXME: k = 1 (or 2, 3) doesn't work !
         for k in &[10, 21] {
             let pubkeys: Vec<Concrete<bitcoin::PublicKey>> =
                 keys.iter().map(|pubkey| Concrete::Key(*pubkey)).collect();
             let big_thresh = Concrete::Threshold(*k, pubkeys);
             let big_thresh_ms: SegwitMiniScript = big_thresh.compile().unwrap();
-            let big_thresh_ms_expected = ms_str!(
+            if *k == 21 {
+                assert!(
+                    matches!(big_thresh_ms.node, Terminal::AndV { .. } | Terminal::Check { .. }),
```

Comment:
This test is a bit hacky but I had no idea how to test it an other way as the compilation is not deterministic. Open to suggestion

Reply:
We don't try to match the exact Miniscripts generated by the compilations. We only check whether the expected satisfaction costs are the same.

Reply:
But that's what other tests do ? (I mostly used `compile_misc()` as an example)

Ok so I kept the miniscript assertion for those that compiled straight (thresh => multi and thresh => thresh), and added script_size assertion on all.